app/src/main/java/emu/skyline/preference/IntegerListPreference.kt

```diff
@@ -0,0 +1,329 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+package emu.skyline.preference
+
+import android.os.Bundle
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import android.util.AttributeSet
+import android.content.Context
+import android.content.res.Resources
+import android.content.res.TypedArray
+import android.content.DialogInterface
+import android.annotation.SuppressLint
+import androidx.annotation.ArrayRes
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.res.TypedArrayUtils
+import androidx.preference.R
+import androidx.preference.DialogPreference
+import androidx.preference.PreferenceDialogFragmentCompat
+import emu.skyline.R as sR
+import emu.skyline.di.getSettings
+
+/**
+ * A Preference that displays a list of entries as a dialog.
+ * This preference saves an integer value instead of a string one.
```

Comment:
I would write this as `A Preference that stores the index of the selected entry from a list of strings`

Reply:
That's not really accurate as an integer type entryValues array can be supplied and it would not save the index in that case.